### PR TITLE
Company slug names are now validated

### LINKED ISSUE #541: Company: Validate slugs on company creation
Currently you can create a company with an invalid slug, after which it is impossible to edit it because the company admin also uses slugs for editing companies (why?).

### PR DESCRIPTION
**Pull request by wsinnema**
They are checked for uniqueness and forbidden characters. Fixed #541

### CODE REVIEW

File: module/Company/src/Company/Form/EditCompany.php [hunk 1]
```diff
@@ -205,6 +205,24 @@ protected function initFilters()
...
+                    'fields' => 'slugName',
+                ]),
+                new \Zend\Validator\Regex([
+                        'pattern' => '/^[0-9a-zA-Z_\.\-]*$/',
+                    ]
+                ),
+            ],
+            'filters' => [
+                ['name' => 'StripTags'],
+                ['name' => 'StringTrim'],
```
Comment (jszanto) [inline module/Company/src/Company/Form/EditCompany.php]: Why are these filters needed when you're already validating the slug properly?

Comment (wsinnema) [inline module/Company/src/Company/Form/EditCompany.php]: No, they are not needed

**Comment by wsinnema** [comment]
There eas still a code climate issue

**Comment by jszanto** [comment]
Oops, I missed that

**Pull request merged**